src/base/dof_map_constraints.C

```diff
@@ -3346,6 +3346,22 @@ void DofMap::process_constraints (MeshBase & mesh)
     check_for_constraint_loops();
   }
 
+  // Adjoints will be constrained where the primal is
+  // Therefore, we will expand the adjoint_constraint_values
+  // map whenever the primal_constraint_values map is expanded
+
+  // First, figure out the total number of QoIs
+  const unsigned int max_qoi_num =
+    _adjoint_constraint_values.empty() ?
+    0 : _adjoint_constraint_values.rbegin()->first+1;
+
+  // A vector to hold pointers to the adjoint constraints map for each adjoint variable
+  std::vector<DofConstraintValueMap *> _adjoint_constraints_maps;
+  _adjoint_constraints_maps.resize(max_qoi_num);
+
+  for ( unsigned int j = 0; j < max_qoi_num; j++)
+    _adjoint_constraints_maps[j] = &(_adjoint_constraint_values.find(j)->second);
```

Comment:
All is madness!

This is an obvious bug whenever find() returns end(), but not only did I not see that before, the code worked fine in CI for me too.  I'm not sure what about my OS or compiler or phase of the moon has changed since, but I get crashes in 4 or 5 GRINS tests now when the later code here tries to dereference end().  I'll fix it.